tests/integration_tests/tests/agentless_tests/test_snapshot.py

```diff
@@ -446,41 +446,38 @@ def _upload_and_restore_snapshot(
         """Upload the snapshot and launch the restore workflow
         """
         snapshot_id = snapshot_id or self.SNAPSHOT_ID
-        rest_client = utils.create_rest_client(
-            tenant=tenant_name, host=self.env.container_ip)
-        self._upload_and_validate_snapshot(snapshot_path,
-                                           snapshot_id,
-                                           rest_client)
-        self.logger.debug('Restoring snapshot...')
-        rest_client.maintenance_mode.activate()
-        execution = rest_client.snapshots.restore(
-            snapshot_id,
-            ignore_plugin_failure=ignore_plugin_failure)
-        # give the database some time to downgrade/upgrade before running
-        # requests to avoid the deadlock described in CY-1455
-        time.sleep(10)
-        self.wait_for_snapshot_restore_to_end(execution.id, client=rest_client)
-        rest_client.maintenance_mode.deactivate()
-        execution = self._wait_for_restore_execution_to_end(
-            execution, rest_client, timeout_seconds=240)
+        with self.client_using_tenant(self.client, tenant_name):
+            self._upload_and_validate_snapshot(snapshot_path,
+                                               snapshot_id)
+            self.logger.debug('Restoring snapshot...')
+            self.client.maintenance_mode.activate()
+            execution = self.client.snapshots.restore(
+                snapshot_id,
+                ignore_plugin_failure=ignore_plugin_failure)
+            # give the database some time to downgrade/upgrade before running
+            # requests to avoid the deadlock described in CY-1455
+            time.sleep(10)
```

Comment:
Is there some sort of actual polling check we can reasonably do for this instead of an arbitrary sleep?

Reply:
we'd have to look at the db directly i think; maybe so, but for now i want to just fix the tests, not rewrite them